assets/check

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python3
+
+import boto3
+import botocore
+import json
+import re
+import sys
+from datetime import datetime
+
+DEFAULT_AWS_REGION = "eu-west-2"
+REQ_METHODS = [
+    "ec2_tags",
+    "volume_tags",
+    "volume_id"
+]
+REQ_PARAMS = [
+    "aws_access_key_id",
+    "aws_secret_access_key",
+    "aws_region",
+    "lookup_method"
+]
+TAGS_REGEX = '^(?:[0-9a-zA-Z-]*=[0-9a-zA-Z-]*(?:,|$))+$'
+VOLUMEID_REGEX = '^vol-[0-9a-f]{17}$'
+
+def set_source_defaults(payload):
+    if "aws_region" not in payload["source"].keys():
+        payload["source"].update({"aws_region": DEFAULT_AWS_REGION})
+
+    return payload
+
+
+def validate_source(payload):
+    for param in REQ_PARAMS:
+        if param not in payload["source"].keys():
+            sys.stderr.write("\nRequired parameter {} not defined\n".format(param))
+            sys.exit(1)
+            
+    if payload["source"]["lookup_method"] not in REQ_METHODS:
+        sys.stderr.write("\nInvalid lookup method; must be one of {}\n".format(REQ_METHODS))
+        sys.exit(1)
+    else:
+        REQ_PARAMS.append(payload["source"]["lookup_method"])
+        if payload["source"]["lookup_method"] == "ec2_tags":
+          REQ_PARAMS.append("ec2_devicename")
+
+    for param in REQ_PARAMS:
+        if param not in payload["source"].keys():
+            sys.stderr.write("\nRequired parameter {} not defined\n".format(param))
+            sys.exit(1)
+
+
+def create_ec2_client(payload):
+    try:
+        session = boto3.Session(
+            aws_access_key_id=payload["source"]["aws_access_key_id"],
+            aws_secret_access_key=payload["source"]["aws_secret_access_key"],
+            region_name=payload["source"]["aws_region"]
+        )
+        ec2_client = session.client(
+            "ec2",
+            payload["source"]["aws_region"]
+        )
+        return ec2_client
+
+    except botocore.exceptions.ClientError as err:
+        sys.stderr.write("Failed to connect to AWS:\n{}".format(err))
+        sys.exit(1)
+
+
+def validate_response(response):
+    if response["ResponseMetadata"]["HTTPStatusCode"] == 200:
```

Comment:
Do we need to return a value? If we simply error and exit upon an invalid response we can just invoke this method elsewhere without any conditional logic safe in the assumption that if it's invalid it will have errored and exited? E.g.

```
if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
    sys.stderr.write(
        "Received {} error from AWS".format(
            response["ResponseMetadata"]["HTTPStatusCode"]
        )
    )
    sys.exit(1)
```

Reply:
Updated in [b1de7a3](https://github.com/companieshouse/ci-resource-type-snapshot-id/pull/1/commits/b1de7a3ccf18a98174ecde9a4ffa6325015d2c0e)